ssl/ssl_asn1.c

```diff
@@ -352,6 +361,15 @@ SSL_SESSION *d2i_SSL_SESSION(SSL_SESSION **a, const unsigned char **pp,
         ret->ext.alpn_selected_len = 0;
     }
 
+    if (as->tick_nonce != NULL) {
+        ret->ext.tick_nonce = as->tick_nonce->data;
```

Comment:
Most of the code above does a strndup; should this be a memdup-type of operation, rather than stealing the pointer from the ASN1_STRING?